src/SprykerEco/Zed/Payone/Business/Payment/MethodMapper/CashOnDelivery.php

```diff
@@ -0,0 +1,228 @@
+    {
+        $bankAccountCheckContainer = new BankAccountCheckContainer();
+
+        $bankAccountCheckContainer->setAid($this->getStandardParameter()->getAid());
+        $bankAccountCheckContainer->setBankCountry($bankAccountCheckTransfer->getBankCountry());
+        $bankAccountCheckContainer->setBankAccount($bankAccountCheckTransfer->getBankAccount());
+        $bankAccountCheckContainer->setBankCode($bankAccountCheckTransfer->getBankCode());
+        $bankAccountCheckContainer->setIban($bankAccountCheckTransfer->getIban());
+        $bankAccountCheckContainer->setBic($bankAccountCheckTransfer->getBic());
+        $bankAccountCheckContainer->setLanguage($this->getStandardParameter()->getLanguage());
+
+        return $bankAccountCheckContainer;
+    }
+
+    /**
+     * @param \Orm\Zed\Payone\Persistence\SpyPaymentPayone $paymentEntity
+     * @param \SprykerEco\Zed\Payone\Business\Api\Request\Container\Authorization\AbstractAuthorizationContainer $authorizationContainer
+     *
+     * @return \SprykerEco\Zed\Payone\Business\Api\Request\Container\Authorization\AbstractAuthorizationContainer
+     */
+    protected function mapPaymentToAbstractAuthorization(
+        SpyPaymentPayone $paymentEntity,
+        AbstractAuthorizationContainer $authorizationContainer
+    ): AbstractAuthorizationContainer {
+        $paymentDetailEntity = $paymentEntity->getSpyPaymentPayoneDetail();
+
+        $authorizationContainer->setAid($this->getStandardParameter()->getAid());
+        $authorizationContainer->setClearingType(PayoneApiConstants::CLEARING_TYPE_CASH_ON_DELIVERY);
+        $authorizationContainer->setReference($paymentEntity->getReference());
+        $authorizationContainer->setAmount($paymentDetailEntity->getAmount());
+        $authorizationContainer->setCurrency($this->getStandardParameter()->getCurrency());
+        $authorizationContainer->setPaymentMethod($this->createPaymentMethodContainerFromPayment($paymentEntity));
+
+        $personalContainer = new PersonalContainer();
+        $this->mapBillingAddressToPersonalContainer($personalContainer, $paymentEntity);
+        $personalContainer->setLanguage($this->getStandardParameter()->getLanguage());
+
+        $authorizationContainer->setPersonalData($personalContainer);
+
+        return $authorizationContainer;
+    }
+
+    /**
+     * @param \Orm\Zed\Payone\Persistence\SpyPaymentPayone $paymentEntity
+     *
+     * @return \SprykerEco\Zed\Payone\Business\Api\Request\Container\Authorization\PaymentMethod\CashOnDeliveryContainer
+     */
+    protected function createPaymentMethodContainerFromPayment(SpyPaymentPayone $paymentEntity): CashOnDeliveryContainer
+    {
+        $shippingProviderName = $paymentEntity->getSpyPaymentPayoneDetail()->getShippingProvider();
+
+        $translatedShippingProviderName = $shippingProviderName ? $this->translate($shippingProviderName) : $shippingProviderName;
```

Comment:
So we might have no shipping provider? Are you sure that in this case we should just continue?

Reply:
We will receive Flash message from API in this case, which says `invalid shipment provider`